### ISSUE TITLE
documentation for creating packages and installers

### ISSUE THREAD

**Issue by nickerso** [cellml/libcellml#503]
#494 adds the CMake/CPack framework for creating various packages and installers for libCellML. The use, configuration, environment setup for using these needs to be documented. Possibly as part of defining tests for them (#502)?

**Comment by agarny** [comment]
FWIW, on macOS, we shouldn't really distribute a `.pkg` file, but a disk image (i.e. a `.dmg` file). For this, all we need to do is to run the following command:

```
hdiutil create -srcfolder installer.pkg installer.dmg
```

**Comment by hsorby** [comment]
What reason do you have for not distributing a '.pkg' file?

**Comment by agarny** [comment]
Ok, I can see that it has been a long time since I have looked into `.pkg` vs. `.dmg`.

Basically, I was recommending `.dmg` because it was *the* way for many years and that was because a `.pkg` file was actually a folder. So, it made sense (and was even advised) to distribute a `.pkg` "file" as a `.dmg` file.

However, nowadays, `.pkg` files are proper files, which means that there is no valid reason for distributing a `.dmg` file that contains only one `.pkg` file.

So, in summary:
 - Thanks for "forcing" me to look into it! :)
 - We can safely distribute a `.pkg` file (nowadays) and therefore don't need to do that through a `.dmg` file.

**Comment by hsorby** [comment]
Great `.pkg` it is!
